src/make_static.h

```diff
@@ -17,11 +17,16 @@
  *    // is an equivalent of:
  *    static const V v = expr;
  *
+ *    Note: `expr` being char* (i.e. "a string literal") is a special case:
+ *      for such type of argument,  STATIC(expr) has `const std::string &` result type.
+ *      This is to support expressions like STATIC("...") instead of STATIC(std::string("..."))
  */
 #define STATIC(expr) \
-    ([]()-> const auto &{ \
-        static const auto __ret = expr; \
+    (([]()-> const auto &{ \
+        using __T = std::conditional_t<std::is_same<std::decay_t<decltype(expr)>, const char*>::value, \
```

Comment:
Names containing a double underscore are reserved for the implementation. Please don't use them here.  (Similarly names starting with an underscore followed by a capital letter).

Reply:
Changed names to avoid using reserved identifiers.